internal/audit/auditlogtextformatter.go

```diff
@@ -153,47 +155,54 @@ func groupAndPrint(vulnerable []types.Coordinate, nonVulnerable []types.Coordina
-		auditedEntries := entry.Data["audited"].([]types.Coordinate)
-		invalidEntries := entry.Data["invalid"].([]types.Coordinate)
-		packageCount := entry.Data["num_audited"].(int)
-		numVulnerable := entry.Data["num_vulnerable"].(int)
+	if auditedEntries != nil && invalidEntries != nil && vulnerableEntries != nil && buildVersion != nil {
+		auditedEntries := entry.Data["audited"].(map[string]ossIndexTypes.Coordinate)
+		vulnerableEntries := entry.Data["vulnerable"].(map[string]types.Projects)
+		invalidEntries := entry.Data["invalid"].([]ossIndexTypes.Coordinate)
 
 		var sb strings.Builder
 
 		w := tabwriter.NewWriter(&sb, 9, 3, 0, '\t', 0)
 		_ = w.Flush()
 
 		logInvalidSemVerWarning(&sb, f.NoColor, f.Quiet, invalidEntries)
-		nonVulnerablePackages, vulnerablePackages := splitPackages(auditedEntries)
 
-		groupAndPrint(vulnerablePackages, nonVulnerablePackages, f.Quiet, f.NoColor, &sb)
+		groupAndPrint(vulnerableEntries, auditedEntries, f.Quiet, f.NoColor, &sb)
 
 		au := aurora.NewAurora(!f.NoColor)
+
+		var updates []string
+		for _, v := range vulnerableEntries {
+			if v.UpdateCoordinate.Coordinates != "" && v.Update != nil {
+				var issueTitles []string
+				for _, v := range v.Coordinate.Vulnerabilities {
+					issueTitles = append(issueTitles, v.Cve)
+				}
+				comment := au.Green(fmt.Sprintf("// fix issues: %s in %s %s\n", strings.Join(issueTitles, ", "), v.Name, v.Version)).String()
+				replace := au.Blue(fmt.Sprintf("replace %s => %s %s\n\n", v.Update.Path, v.Update.Path, v.Update.Version)).String()
+				updates = append(updates, comment, replace)
+			}
+		}
+
+		if len(updates) > 0 {
+			sb.WriteString(au.Bold("I found some updated versions you can try out, that have no known vulnerabilities!\n\nTry the following in your go.mod file:\n\n").String())
+			for _, v := range updates {
+				sb.WriteString(au.Italic(v).String())
+			}
+		}
+
 		t := table.NewWriter()
 		t.SetStyle(table.StyleBold)
 		t.SetTitle("Summary")
-		t.AppendRow([]interface{}{"Audited Dependencies", strconv.Itoa(packageCount)})
+		t.AppendRow([]interface{}{"Audited Dependencies", strconv.Itoa(len(auditedEntries))})
 		t.AppendSeparator()
-		t.AppendRow([]interface{}{"Vulnerable Dependencies", au.Bold(au.Red(strconv.Itoa(numVulnerable)))})
+		t.AppendRow([]interface{}{"Vulnerable Dependencies", au.Bold(au.Red(strconv.Itoa(len(vulnerableEntries))))})
 		sb.WriteString(t.Render())
 		sb.WriteString("\n")
 
 		return []byte(sb.String()), nil
 	}
 	return nil, errors.New("fields passed did not match the expected values for an audit log. You should probably look at setting the formatter to something else")
 }
-
-func splitPackages(entries []types.Coordinate) (nonVulnerable []types.Coordinate, vulnerable []types.Coordinate) {
-	for _, v := range entries {
-		if v.IsVulnerable() {
-			vulnerable = append(vulnerable, v)
-		} else {
-			nonVulnerable = append(nonVulnerable, v)
-		}
-	}
-	return
-}
```

Comment:
This wasn't even being used, or rather it was, but we already had two lists of data. I removed it because LOL